esinstall/src/index.ts

```diff
@@ -341,17 +335,218 @@ export async function install(
+    splitting: true, // needed to dedupe packages
+    external: externalPackage,
+    minifyIdentifiers: Boolean(treeshake),
+    minifySyntax: Boolean(treeshake),
+    minifyWhitespace: Boolean(treeshake),
+    mainFields: ['browser:module', 'module', 'browser', 'main'].filter(isTruthy),
+    // sourcemap: 'inline', // TODO sourcemaps panics and gives a lot of CPU load
+    define: {
+      'process.env.NODE_ENV': JSON.stringify('dev'),
+      process: 'window', // TODO temporary workarounds to make certain packages work
+      'process.env': 'window',
+      global: 'window',
+      ...generateEnvReplacements(env),
+    },
+    tsconfig: tsconfigTempFile,
+    bundle: true,
+    format: 'esm',
+    write: true,
+    entryPoints,
+    outdir: destLoc,
+    minify: Boolean(treeshake),
+    logLevel: 'info',
+    metafile,
+  });
+
+  await fs.promises.unlink(tsconfigTempFile);
+
+  const meta = JSON.parse(await (await fs.promises.readFile(metafile)).toString());
+
+  const importMap = metafileToImportMap({
+    installEntrypoints,
+    meta,
+    destLoc: destLoc,
+  });
+
+  const stats = metafileToStats({meta, destLoc});
+
+  return {stats, importMap};
+}
+
+function metafileToImportMap(_options: {
+  installEntrypoints: Record<string, string>;
+  meta: Metadata;
+  destLoc: string;
+}): ImportMap {
+  const {destLoc: destLoc, installEntrypoints: installEntrypoints, meta} = _options;
+  const inputFiles = Object.values(installEntrypoints).map((x) => path.resolve(x)); // TODO replace resolve with join in cwd
+  const inputFilesToSpecifiers = invert(installEntrypoints);
+
+  const importMaps: Record<string, string>[] = Object.keys(meta.outputs).map((output) => {
+    // chunks cannot be entrypoints
+    if (path.basename(output).startsWith('chunk.')) {
+      return {};
+    }
+    const inputs = Object.keys(meta.outputs[output].inputs).map((x) => path.resolve(x)); // TODO replace resolve with join in cwd
+    const input = inputs.find((x) => inputFiles.includes(x));
+    if (!input) {
+      return {};
+    }
+    const specifier = inputFilesToSpecifiers[input];
+    return {[specifier]: './' + toUnixPath(path.normalize(path.relative(destLoc, output)))};
+  });
+  const importMap = Object.assign({}, ...importMaps);
+  return {imports: importMap};
+}
+
+function metafileToStats(_options: {meta: Metadata; destLoc: string}): DependencyStatsOutput {
+  const {meta, destLoc} = _options;
+  const stats = Object.keys(meta.outputs).map((output) => {
+    const value = meta.outputs[output];
+    // const inputs = meta.outputs[output].bytes;
+    return {
+      path: output,
+      isCommon: ['chunk.'].some((x) => path.basename(output).startsWith(x)),
+      bytes: value.bytes,
+    };
+  });
+
+  function makeStatObject(value) {
+    const relativePath = toUnixPath(path.relative(destLoc, value.path));
+    return {
+      [relativePath]: {
+        size: value.bytes,
+        gzip: 0, // TODO do we want to waste time compressing to show stats?
```

Comment:
Just my not-very-strongly-held opinion: I don’t really get anything out of showing gzip or other compressed sizes when building (hot take: I think they’re used as a smokescreen to justify too much JS in your build). I think the unpacked size is more significant because that’s what gets parsed & executed. So I’d be fine only showing uncompressed size.

Reply:
Gzip information is useful to indicate the real network load.